docs/counting.md

```diff
@@ -0,0 +1,223 @@
+offers a bit more functionality... some extra sugar if you will. Let's take a
+look.
+
+#### Creating a counter
+
+Usually, a counter is created with the {@link RedisMetrics#counter} function.
+Let's start by creating a counter that is capable of reporting counts with a
+time granularity of one hour:
+
+```javascript
+var metrics = require('redis-metrics')();
+var myCounter = metrics.counter('pageview', {
+  timeGranularity: 'hour',
+  expireKeys: true
+};
+```
+
+Here, we are passing an options object explicitly to the counter. This is
+useful when we want different settings for different counters. We can also
+specify some default counter settings on the metrics module itself:
+
+```javascript
+var metrics = require('redis-metrics')({
+  counterOptions: {
+    timeGranularity: 'hour',
+    expireKeys: true
+  }
+});
+
+// Will use the counterOptions object because no options are provided.
+var myCounter = metrics.counter('pageview');
+```
+
+#### Incrementing a counter
+
+The counter can be incremented in two ways, using {@link
+TimestampedCounter#incr|incr} or {@link TimestampedCounter#incrby|incrby}:
+
+```javascript
+// Increment by 1
+myCounter.incr(function(err, result) {
+  console.log(result);
+});
+
+// Increment by 5
+myCounter.incrby(5, function(err, result) {
+  console.log(result);
+});
+```
+
+The `err` and `result` parameters contain the answer from Redis.
+
+This looks and feels as if we are incrementing a normal Redis counter. Behind
+the scenes, that is exactly what is happening but with a slight twist. Since we
+have chosen a time granularity of an hour, more than one counter is being
+incremented for various timestamps. This does give some extra overhead, but it
+makes it very easy and fast to retrieve a count for a specific period.
+
+#### Fetching a count
+
+Since our counter has a time granularity of an hour, we can now answer
+questions such as: "How many page views do I have so far today?" and "How many
+page views do I have so far this hour?":
+
+```javascript
+// Number of page views today.
+myCounter.count('day', function(err, result) {
+  console.log(result);
+});
+
+// Number of page views this hour.
+myCounter.count('hour', function(err, result) {
+  console.log(result);
+});
+```
+
+Of course, we can also get the total number of page views:
+
+```javascript
+// Number of page views in total.
+myCounter.count(function(err, result) {
+  console.log(result);
+});
+```
+
+In all of the above methods `result` is single integer.
+
+#### Fetching a count for a time range
+
+Fetching a single count for the current time is useful, but often it makes
+sense to view metrics over a period of time. Let's ask the counter "What is my
+page view count for the last 24 hours until now?" using the {@link
+TimestampedCounter#countRange|countRange} function:
+
+```javascript
+// Moment is nice and easy for dates so we'll just use that here.
+var moment = require('moment');
+var yesterday = moment().add(-24, 'hours');
```

Comment:
Is there a reason why you used add -24 instead of subtract 24 or subtract 1 day?